app/viewsapi/FormFunctions.scala

```diff
@@ -13,6 +16,28 @@ object FormFunctions {
       if (args.exists(_ == (requiredInputArg, true))) "*" else ""
     }
   }
+  class ErrorHandling(elements: FieldElements) {
+
+    def setErrorClass(): String = {
+      if(elements.hasErrors) {
+        "govuk-form-group--error"
+      } else {
+        ""
+      }
+    }
+
+    def renderErrorMessage(): Html = {
+      if(elements.hasErrors) {
+        Html("<span id=\"error\" class=\"govuk-error-message\"><span class=\"govuk-visually-hidden\">Error:</span>" +
```

Comment:
I'm a bit wary of having HTML in the Scala classes. I know it's a choice between having HTML in the Scala, or logic in the templates, but I think having some logic is OK if it's very simple like an if statement.